src/internet_identity/src/main.rs

```diff
@@ -766,7 +768,8 @@ mod attribute_sharing_mvp {
     async fn prepare_id_alias(
         req: PrepareIdAliasRequest,
     ) -> Result<PreparedIdAlias, PrepareIdAliasError> {
-        let _maybe_ii_domain = authenticate_and_record_activity(req.identity_number);
+        authenticate_and_record_activity(req.identity_number)
+            .map_err(|_err| PrepareIdAliasError::Unauthorized)?;
```

Comment:
The `PrepareIdAliasError` will be made consistent with `principal` information and the `InternalCanisterError` variant in an upcoming PR.